### PR TITLE
Use js-utility-belt

### PR DESCRIPTION
**Pull request by sohkai**
Removes the `client/lib/utils` and replaces them with imports from `js-utility-belt`.

Also contains a few fixes to the app as well as some docker settings.

### CODE REVIEW

File: client/lib/js/constants/api_urls.js [hunk 1]
```diff
@@ -4,11 +4,11 @@ import { API_PATH } from './application_constants';
...
     'accounts': API_PATH + 'accounts/',
-    'accounts_detail': API_PATH + 'accounts/${account_id}/',
-    'assets_for_account': API_PATH + 'accounts/${account_id}/assets/',
+    'accounts_detail': API_PATH + 'accounts/%(accountId)s/',
+    'assets_for_account': API_PATH + 'accounts/%(accountId)s/assets/',
     'assets': API_PATH + 'assets/',
-    'assets_detail': API_PATH + 'assets/${asset_id}/${cid}/',
-    'assets_transfer': API_PATH + 'assets/${asset_id}/${cid}/transfer/'
+    'assets_detail': API_PATH + 'assets/%(assetId)s/',
+    'assets_transfer': API_PATH + 'assets/%(assetId)s/%(cid)s/transfer/'
```
Comment (diminator) [inline client/lib/js/constants/api_urls.js:11]: this is the string formatter in JS?

File: client/lib/js/utils/request.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+    }
+};
+
+/**
+ * Small wrapper around js-utility-belt's request that provides default settings and response
+ * handling
+ */
+export default function request(url, config) {
```
Comment (diminator) [inline client/lib/js/utils/request.js:18]: would it make sense to provide this as a basic interface in js-utility-belt?

**Comment by diminator** [comment]
awesome work @SohKai!

tried it out and works like a charm

:+1:

File: client/lib/js/constants/api_urls.js (hunk 1 above)
Comment (sohkai) [inline client/lib/js/constants/api_urls.js:11]: There isn't really a built-in recommend string formatter in JS aside from rolling your own regex. This is the format used by [sprintf-js](https://github.com/alexei/sprintf.js) that's a close approximation to sprintf; it's basically a super powerful version of our old `formatText`.

File: client/lib/js/utils/request.js (hunk 2 above)
Comment (sohkai) [inline client/lib/js/utils/request.js:18]: I'd shy away from it since it really loses flexibility very quickly if we do things like assuming json responses and stripping null query parameters.

Comment (diminator) [inline client/lib/js/utils/request.js:18]: ok

Op di 31 mei 2016 om 14:15 schreef Brett Sun notifications@github.com:

File: client/lib/js/constants/api_urls.js (hunk 1 above)
Comment (sohkai) [inline client/lib/js/constants/api_urls.js:11]: We can now revert these changes to `ApiUrls` with https://github.com/bigchaindb/js-utility-belt/pull/12/.
